Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -146,7 +149,16 @@
     /**
      * An implementation of forest traversal algorithms.
      */
-    protected final AbstractForestTraversalExecutor executor;
+    protected final AbstractForestTraversalExecutor traversalExecutor;
```

Comment:
Do you have an alternative approach. I agree with a previous discussion that we should have a clear separation between forest state and executor classes, but in order for `forest.getAnomalyScore()` to work the forest will need an executor instance. Are you suggesting a different organization of classes where a forest is purely the model/data class and the `getAnomalyScore()` method (or its equivalent) would live somewhere else?

Reply:
here's my train of thoughts.
what i see is that AbstractForestTraversalExecutor contains trees. The question is does the data need to be in an executor? or can the executor just take the trees managed elsewhere as input so data can managed closely together without executor in the way?

so another idea may look like this.
1. data under executor (trees) is moved to maybe under forest directly. so there is no executor layer between forest and trees. the benefits are a. accessing data doesn't require executor. b. executor can better focus on operations on data rather than taking care of the data, and c. data of a model is closer together.
2. executor can still be used to run on the data by taking the data as input.
3. the forest-level api like getAnomalyScore does not need to change since internally executor is still used on the model data.
4. the executor can then be changed more freely without worrying about affecting data. a few examples:
a) be able to update executor config at any time, current to/from thread pool, 10 threads to/from 100 threads
b) be able to use given thread pool rather than creating internally. maybe replace current thread config api with directly taking an optional thread pool (default is empty meaning current thread), which can be reset later. Then the same thread pool can be shared across multiple models and other tasks. After all, i don't think creating a thread pool is a necessary responsibility for rcf/ml models. Taking one from config input is not hard for client but makes it explicit and clear to client.
c) be able to choose whether thread config is included in serialization. So computational config is only an option secondary to main model data.

Reply:
I am comfortable with "components" as sub pieces of RCF. I'm not sure if I'd call each of the components a "model" -- because for tasks like imputation we need more than one "model". But I have no strong opinion re this.